fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/codec/LuceneOptimizedCompoundFormat.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * LuceneOptimizedCompoundFormat.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.lucene.codec;
+
+import org.apache.lucene.codecs.CompoundFormat;
+import org.apache.lucene.codecs.lucene50.Lucene50CompoundFormat;
+import org.apache.lucene.index.IndexFileNames;
+import org.apache.lucene.index.SegmentInfo;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.IOContext;
+
+import java.io.IOException;
+import java.util.stream.Collectors;
+
+/**
+ *
+ * Wrapper for the Lucene50CompoundFormat to optimize compound files for sitting on FoundationDB.
```

Comment:
Might be good to document here (or point to the `LuceneOptimizedCodec` javadoc) what the differences between this format and the standard format is.

Reply:
Added one line in the java doc: ```In contrast to {@link Lucene50CompoundFormat}, this wrapper uses {@link LuceneOptimizedWrappedDirectory}
 * as the {@link Directory} for read/write of compound file.```